x-pack/platform/plugins/shared/streams/server/lib/content/stream.ts

```diff
@@ -175,3 +172,22 @@ export function resolveAncestors(leafs: string[]) {
   const streamNames = leafs.flatMap((name) => getAncestorsAndSelf(name));
   return uniq(streamNames);
 }
+
+function assertNoConflictingRouting(
+  rootRouting: RoutingDefinition[],
+  importedRouting: RoutingDefinition[]
+) {
+  for (const { destination } of importedRouting) {
+    if (rootRouting.some((rule) => rule.destination === destination)) {
+      throw new ContentPackConflictError(`Child stream [${destination}] already exists`);
+    }
+  }
+}
+
+function assertNoConflictingFields(rootFields: FieldDefinition, importedFields: FieldDefinition) {
+  for (const [field, { type }] of Object.entries(importedFields)) {
```

Comment:
Fields can have more stuff in them than just type, we should probably do a deep equality check: https://github.com/elastic/kibana/blob/440aeabd398d3ca29e578f076a3292f5a61ec217/x-pack/platform/packages/shared/kbn-streams-schema/src/fields/index.ts#L27